test(dtslint): add zipAll

timdeschryver opened the pull request:
Description:
This PR adds dtslint tests for `zipAll`.

Related issue (if exists): #4093

cartant commented on spec-dtslint/operators/zipAll-spec.ts:
```diff
@@ -0,0 +1,25 @@
+
+it('should support custom typings on the projector', () => {
+  const o = of(of(1, 2, 3)).pipe(zipAll<string>(value => value)); // $ExpectType Observable<string>
+  const p = of(of(1, 2, 3)).pipe(zipAll<string>((value: string) => value)); // $ExpectType Observable<string>
```
If this test passes, it looks to me like there must be an error in the typings - there is no `string` in the source observable.

timdeschryver replied:
I sure can 😄 !

benlesh replied:
this might actually be a problem with `pipe` typings, not `zipAll`. :thinking:  I think we can just remove these tests for now and file a separate issue.

cartant commented:
Yeah, that looks wrong, too. I'll have a look at this later - after I've had a coffee. Perhaps the signatures are there due to issues that arise with early versions of TypeScript? Anyway, I think the `any` signatures should be removed.

timdeschryver commented:
Woops, seems like this wasn't as simple as I thought 😅 
I removed the `any` typings and the existing tests started to fail.
I reverted my last commit and will take a look at it later.

timdeschryver commented:
Is this good to merge? Or is there something I should address?

benlesh commented on spec-dtslint/operators/zipAll-spec.ts:
```diff
@@ -0,0 +1,24 @@
+});
+
+it('should enforce types', () => {
+  const o = of(1, 2, 3).pipe(zipAll()); // $ExpectError
```
Can we add a couple more here for other types of Observable inputs? Iterators, Promises, Arrays, etc.

timdeschryver replied:
Forgot about this, here you go 😅 
I had to remove the following check

```ts
const p = of(of(1, 2, 3)).pipe(zipAll((value: string) => value)); // $ExpectError
```
